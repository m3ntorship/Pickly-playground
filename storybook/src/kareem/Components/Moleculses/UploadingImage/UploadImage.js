import React from "react";
import Photo from "../../Atoms/Photos/Photo";
import InputImage from "../../Atoms/Inputtexts/InputImage";

function UploadImage({ status, imageSrc, fullRound, width, icon }) {
  return (
    <div>
      <Photo status={status} imageSrc={imageSrc} />
      <InputImage fullRound={fullRound} width={width} icon={icon} />
    </div>
  );
}

export default UploadImage;
