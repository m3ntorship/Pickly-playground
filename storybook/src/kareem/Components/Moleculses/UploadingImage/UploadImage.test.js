import React from "react";
import renderer from "react-test-renderer";
import UploadImage from "./UploadImage";
//status, imageSrc, fullRound, width, icon
describe("Render the UploadImage component", () => {
  it("Kareem: Render UploadImage in status normal ", () => {
    const tree = renderer
      .create(
        <UploadImage
          status="normal"
          imageSrc="normal"
          fullRound={false}
          width="704px"
          icon="A"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Kareem: Render UploadImage in loading case ", () => {
    const tree = renderer
      .create(
        <UploadImage
          status="loading"
          imageSrc="normal"
          fullRound={false}
          width="704px"
          icon="A"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Kareem: Render UploadImage in notload case ", () => {
    const tree = renderer
      .create(
        <UploadImage
          status="notload"
          imageSrc="normal"
          fullRound={false}
          width="704px"
          icon="A"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
