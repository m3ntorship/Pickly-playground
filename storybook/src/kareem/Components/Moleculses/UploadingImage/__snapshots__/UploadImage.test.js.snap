// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the UploadImage component Kareem: Render UploadImage in loading case  1`] = `
<div>
  <div
    className="relative items-center"
    height="708px"
    width="704px"
  >
    <img
      alt="opacity-30 relative items-center"
      className="opacity-30 relative items-center"
      height="708px"
      src="./pickly-man.png"
      style={
        Object {
          "Zindex": "z-index: -1",
        }
      }
      width="704px"
    />
    <div
      className=" w-56 h-3  absolute justify-start align-middle border-white rounded-sm top-80 left-60 bg-white"
    >
      <div
        className=" w-40 h-1 bg-primary rounded-sm ml-1 mt-1 "
      />
    </div>
    <div
      className="hidden absolute justify-center align-middle w-96 h-48  top-72 left-40"
    >
      <div
        className="flex-col items-center justify-center align-middle  w-96 h-48   "
      >
        <svg
          className="w-20 h-20 ml-36"
          fill="none"
          viewBox="0 0 80 80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M39.8432 6.66675C21.5498 6.66675 6.6665 21.6201 6.6665 40.0001C6.6665 58.3801 21.6198 73.3334 39.9998 73.3334C58.3798 73.3334 73.3332 58.3801 73.3332 40.0001C73.3332 21.6201 58.3098 6.66675 39.8432 6.66675ZM39.9998 66.6667C25.2965 66.6667 13.3332 54.7034 13.3332 40.0001C13.3332 25.2967 25.2232 13.3334 39.8432 13.3334C54.6365 13.3334 66.6665 25.2967 66.6665 40.0001C66.6665 54.7034 54.7032 66.6667 39.9998 66.6667Z"
            fill="#FE544A"
          />
          <path
            d="M36.6665 23.3333H43.3332V46.6666H36.6665V23.3333ZM36.6665 49.9999H43.3332V56.6666H36.6665V49.9999Z"
            fill="#FE544A"
          />
        </svg>
        <span
          className="h-9 w-96 font-sans text-lg text-error font-bold block mx-4"
        >
          Image couldn't be uploaded!
        </span>
        <span
          className="h-5 font-sans text-sm text-error block mx-32"
        >
          Max size is 2 MB
        </span>
      </div>
    </div>
  </div>
  <div
    className="rounded-bl-md rounded-br-md h-10  border-grey-shd5 border mt-1 mb-12 pr-4 flex align-middle items-center"
    style={
      Object {
        "width": "704px",
      }
    }
  >
    <svg
      className=" ml-2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
        fill="#F2F9FF"
      />
      <path
        d="M14.085 14.4023H9.91504L8.97852 17H7.625L11.4258 7.04688H12.5742L16.3818 17H15.0352L14.085 14.4023ZM10.3115 13.3223H13.6953L12 8.66699L10.3115 13.3223Z"
        fill="#8D9A9E"
      />
    </svg>
    <svg
      className="hidden ml-2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
        fill="#F2F9FF"
      />
      <path
        d="M9.29395 17V7.04688H12.5479C13.6279 7.04688 14.4391 7.27018 14.9814 7.7168C15.5283 8.16341 15.8018 8.82422 15.8018 9.69922C15.8018 10.1641 15.6696 10.5765 15.4053 10.9365C15.141 11.292 14.7809 11.5677 14.3252 11.7637C14.863 11.9141 15.2868 12.2012 15.5967 12.625C15.9111 13.0443 16.0684 13.5456 16.0684 14.1289C16.0684 15.0221 15.779 15.724 15.2002 16.2344C14.6214 16.7448 13.8034 17 12.7461 17H9.29395ZM10.6064 12.3447V15.9268H12.7734C13.3841 15.9268 13.8649 15.7695 14.2158 15.4551C14.5713 15.1361 14.749 14.6986 14.749 14.1426C14.749 12.944 14.0973 12.3447 12.7939 12.3447H10.6064ZM10.6064 11.292H12.5889C13.1631 11.292 13.6211 11.1484 13.9629 10.8613C14.3092 10.5742 14.4824 10.1846 14.4824 9.69238C14.4824 9.14551 14.3229 8.74902 14.0039 8.50293C13.6849 8.25228 13.1995 8.12695 12.5479 8.12695H10.6064V11.292Z"
        fill="#8D9A9E"
      />
    </svg>
    <input
      className="h-5 text-sm font-sans outline-none ml-2 text-grey-shd1"
      id="text2"
      style={
        Object {
          "width": "704px",
        }
      }
      type="text"
    />
  </div>
</div>
`;

exports[`Render the UploadImage component Kareem: Render UploadImage in notload case  1`] = `
<div>
  <div
    className="relative items-center"
    height="708px"
    width="704px"
  >
    <img
      alt="opacity-5 "
      className="opacity-5 "
      height="708px"
      src="./pickly-man.png"
      style={
        Object {
          "Zindex": "z-index: -1",
        }
      }
      width="704px"
    />
    <div
      className="hidden w-56 h-3  absolute justify-start align-middle border-white rounded-sm top-80 left-60 bg-white"
    >
      <div
        className="hidden w-40 h-1 bg-primary rounded-sm ml-1 mt-1 "
      />
    </div>
    <div
      className=" absolute justify-center align-middle w-96 h-48  top-72 left-40"
    >
      <div
        className="flex-col items-center justify-center align-middle  w-96 h-48   "
      >
        <svg
          className="w-20 h-20 ml-36"
          fill="none"
          viewBox="0 0 80 80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M39.8432 6.66675C21.5498 6.66675 6.6665 21.6201 6.6665 40.0001C6.6665 58.3801 21.6198 73.3334 39.9998 73.3334C58.3798 73.3334 73.3332 58.3801 73.3332 40.0001C73.3332 21.6201 58.3098 6.66675 39.8432 6.66675ZM39.9998 66.6667C25.2965 66.6667 13.3332 54.7034 13.3332 40.0001C13.3332 25.2967 25.2232 13.3334 39.8432 13.3334C54.6365 13.3334 66.6665 25.2967 66.6665 40.0001C66.6665 54.7034 54.7032 66.6667 39.9998 66.6667Z"
            fill="#FE544A"
          />
          <path
            d="M36.6665 23.3333H43.3332V46.6666H36.6665V23.3333ZM36.6665 49.9999H43.3332V56.6666H36.6665V49.9999Z"
            fill="#FE544A"
          />
        </svg>
        <span
          className="h-9 w-96 font-sans text-lg text-error font-bold block mx-4"
        >
          Image couldn't be uploaded!
        </span>
        <span
          className="h-5 font-sans text-sm text-error block mx-32"
        >
          Max size is 2 MB
        </span>
      </div>
    </div>
  </div>
  <div
    className="rounded-bl-md rounded-br-md h-10  border-grey-shd5 border mt-1 mb-12 pr-4 flex align-middle items-center"
    style={
      Object {
        "width": "704px",
      }
    }
  >
    <svg
      className=" ml-2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
        fill="#F2F9FF"
      />
      <path
        d="M14.085 14.4023H9.91504L8.97852 17H7.625L11.4258 7.04688H12.5742L16.3818 17H15.0352L14.085 14.4023ZM10.3115 13.3223H13.6953L12 8.66699L10.3115 13.3223Z"
        fill="#8D9A9E"
      />
    </svg>
    <svg
      className="hidden ml-2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
        fill="#F2F9FF"
      />
      <path
        d="M9.29395 17V7.04688H12.5479C13.6279 7.04688 14.4391 7.27018 14.9814 7.7168C15.5283 8.16341 15.8018 8.82422 15.8018 9.69922C15.8018 10.1641 15.6696 10.5765 15.4053 10.9365C15.141 11.292 14.7809 11.5677 14.3252 11.7637C14.863 11.9141 15.2868 12.2012 15.5967 12.625C15.9111 13.0443 16.0684 13.5456 16.0684 14.1289C16.0684 15.0221 15.779 15.724 15.2002 16.2344C14.6214 16.7448 13.8034 17 12.7461 17H9.29395ZM10.6064 12.3447V15.9268H12.7734C13.3841 15.9268 13.8649 15.7695 14.2158 15.4551C14.5713 15.1361 14.749 14.6986 14.749 14.1426C14.749 12.944 14.0973 12.3447 12.7939 12.3447H10.6064ZM10.6064 11.292H12.5889C13.1631 11.292 13.6211 11.1484 13.9629 10.8613C14.3092 10.5742 14.4824 10.1846 14.4824 9.69238C14.4824 9.14551 14.3229 8.74902 14.0039 8.50293C13.6849 8.25228 13.1995 8.12695 12.5479 8.12695H10.6064V11.292Z"
        fill="#8D9A9E"
      />
    </svg>
    <input
      className="h-5 text-sm font-sans outline-none ml-2 text-grey-shd1"
      id="text2"
      style={
        Object {
          "width": "704px",
        }
      }
      type="text"
    />
  </div>
</div>
`;

exports[`Render the UploadImage component Kareem: Render UploadImage in status normal  1`] = `
<div>
  <div
    className="relative items-center"
    height="708px"
    width="704px"
  >
    <img
      alt=""
      className=""
      height="708px"
      src="./pickly-man.png"
      style={
        Object {
          "Zindex": "",
        }
      }
      width="704px"
    />
    <div
      className="hidden w-56 h-3  absolute justify-start align-middle border-white rounded-sm top-80 left-60 bg-white"
    >
      <div
        className="hidden w-40 h-1 bg-primary rounded-sm ml-1 mt-1 "
      />
    </div>
    <div
      className="hidden absolute justify-center align-middle w-96 h-48  top-72 left-40"
    >
      <div
        className="flex-col items-center justify-center align-middle  w-96 h-48   "
      >
        <svg
          className="w-20 h-20 ml-36"
          fill="none"
          viewBox="0 0 80 80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M39.8432 6.66675C21.5498 6.66675 6.6665 21.6201 6.6665 40.0001C6.6665 58.3801 21.6198 73.3334 39.9998 73.3334C58.3798 73.3334 73.3332 58.3801 73.3332 40.0001C73.3332 21.6201 58.3098 6.66675 39.8432 6.66675ZM39.9998 66.6667C25.2965 66.6667 13.3332 54.7034 13.3332 40.0001C13.3332 25.2967 25.2232 13.3334 39.8432 13.3334C54.6365 13.3334 66.6665 25.2967 66.6665 40.0001C66.6665 54.7034 54.7032 66.6667 39.9998 66.6667Z"
            fill="#FE544A"
          />
          <path
            d="M36.6665 23.3333H43.3332V46.6666H36.6665V23.3333ZM36.6665 49.9999H43.3332V56.6666H36.6665V49.9999Z"
            fill="#FE544A"
          />
        </svg>
        <span
          className="h-9 w-96 font-sans text-lg text-error font-bold block mx-4"
        >
          Image couldn't be uploaded!
        </span>
        <span
          className="h-5 font-sans text-sm text-error block mx-32"
        >
          Max size is 2 MB
        </span>
      </div>
    </div>
  </div>
  <div
    className="rounded-bl-md rounded-br-md h-10  border-grey-shd5 border mt-1 mb-12 pr-4 flex align-middle items-center"
    style={
      Object {
        "width": "704px",
      }
    }
  >
    <svg
      className=" ml-2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
        fill="#F2F9FF"
      />
      <path
        d="M14.085 14.4023H9.91504L8.97852 17H7.625L11.4258 7.04688H12.5742L16.3818 17H15.0352L14.085 14.4023ZM10.3115 13.3223H13.6953L12 8.66699L10.3115 13.3223Z"
        fill="#8D9A9E"
      />
    </svg>
    <svg
      className="hidden ml-2"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
        fill="#F2F9FF"
      />
      <path
        d="M9.29395 17V7.04688H12.5479C13.6279 7.04688 14.4391 7.27018 14.9814 7.7168C15.5283 8.16341 15.8018 8.82422 15.8018 9.69922C15.8018 10.1641 15.6696 10.5765 15.4053 10.9365C15.141 11.292 14.7809 11.5677 14.3252 11.7637C14.863 11.9141 15.2868 12.2012 15.5967 12.625C15.9111 13.0443 16.0684 13.5456 16.0684 14.1289C16.0684 15.0221 15.779 15.724 15.2002 16.2344C14.6214 16.7448 13.8034 17 12.7461 17H9.29395ZM10.6064 12.3447V15.9268H12.7734C13.3841 15.9268 13.8649 15.7695 14.2158 15.4551C14.5713 15.1361 14.749 14.6986 14.749 14.1426C14.749 12.944 14.0973 12.3447 12.7939 12.3447H10.6064ZM10.6064 11.292H12.5889C13.1631 11.292 13.6211 11.1484 13.9629 10.8613C14.3092 10.5742 14.4824 10.1846 14.4824 9.69238C14.4824 9.14551 14.3229 8.74902 14.0039 8.50293C13.6849 8.25228 13.1995 8.12695 12.5479 8.12695H10.6064V11.292Z"
        fill="#8D9A9E"
      />
    </svg>
    <input
      className="h-5 text-sm font-sans outline-none ml-2 text-grey-shd1"
      id="text2"
      style={
        Object {
          "width": "704px",
        }
      }
      type="text"
    />
  </div>
</div>
`;
